plugin/plugin.go

```diff
@@ -31,11 +31,25 @@ func stringInSlice(str string, slice []string) bool {
 
 // New returns a new validator plugin.
 func New(privilegedUsers []string, userPermissionsRaw map[string]string) validator.Plugin {
-	userPermissions := make(map[string][]string)
+	userPermissions := make(map[string]map[string][]string)
```

Comment:
What if we model permissions with structs and https://github.com/go-yaml/yaml/tree/v3 or json? That seems better than hacking in our own serialization...

```
map["trent"] = `encoded yaml or json here`
```

Reply:
That's the lack of knowing Go :)
My thought process has been like that:
We have an env variable with text in the config hence we need to parse text.
Is there an easy way to pass yaml or json from the env variable?